src/expr/dtype.cpp

```diff
@@ -143,6 +143,52 @@ size_t DType::cindexOfInternal(Node item)
   return item.getAttribute(DTypeConsIndexAttr());
 }
 
+void DType::getUnresolvedDatatypeTypes(std::set<TypeNode>& unresTypes) const
```

Comment:
Maybe we should coll this `collectUnresolvedDatatypeTypes`, `get` kinda implies that this would return something.

Reply:
Sounds good.